concent_api/api-integration-force-get-task-result-test.py

```diff
@@ -135,65 +150,201 @@ def main():
+        headers={
+            'Content-Type': 'application/octet-stream',
+            'concent-client-public-key': b64encode(REQUESTOR_PUBLIC_KEY).decode('ascii'),
+            'concent-other-party-public-key': b64encode(PROVIDER_PUBLIC_KEY).decode('ascii'),
+        },
+        expected_status  = 200,
+        expected_message = AckForceGetTaskResult
+    )
+    print(f"Waiting {WAIT_TIME_1} seconds...")
+    time.sleep(WAIT_TIME_1)  # TaskToCompute.deadline + FORCE_ACCEPTANCE_TIME < current time <= TaskToCompute.deadline + FORCE_ACCEPTANCE_TIME + CONCENT_MESSAGING_TIME
+    api_request(    # Concent   -> Provider:   ForceGetTaskResult + FileTransferToken
+        cluster_url,
+        'receive',
+        PROVIDER_PRIVATE_KEY,
+        CONCENT_PUBLIC_KEY,
+        headers = {
+            'Content-Type': 'application/octet-stream',
+            'concent-client-public-key': b64encode(PROVIDER_PUBLIC_KEY).decode('ascii')
+        },
+        expected_status  = 200,
+        expected_message = ForceGetTaskResultUpload
+    )
+    print(f"Waiting {CONCENT_MESSAGING_TIME} seconds...")
+    time.sleep(CONCENT_MESSAGING_TIME)  # TaskToCompute.deadline + FORCE_ACCEPTANCE_TIME + CONCENT_MESSAGING_TIME < current time <= TaskToCompute.deadline + FORCE_ACCEPTANCE_TIME + 2 * CONCENT_MESSAGING_TIME.
+    api_request(  # Concent   -> Requestor:  ForceGetTaskResultFailed (provider does not submit anything)
         cluster_url,
         'receive',
         REQUESTOR_PRIVATE_KEY,
         CONCENT_PUBLIC_KEY,
         headers = {
-            'Content-Type':                 'application/octet-stream',
-            'concent-client-public-key':    b64encode(REQUESTOR_PUBLIC_KEY).decode('ascii')
-        }
+            'Content-Type': 'application/octet-stream',
+            'concent-client-public-key': b64encode(REQUESTOR_PUBLIC_KEY).decode('ascii')
+        },
+        expected_status  = 200,
+        expected_message = ForceGetTaskResultFailed
     )
 
 
+def parse_arguments():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("cluster_url")
+    parser.add_argument("tc_patterns", nargs='*')
+    args = parser.parse_args()
+    return args.cluster_url, args.tc_patterns
+
+
+def get_tests_list(patterns, all_objects):
+    def _is_a_test(x):
+        return "case_" in x
+
+    tests = list(filter(lambda x: _is_a_test(x), all_objects))
+    if len(patterns) > 0:
+        safe_patterns = [pattern for pattern in patterns if _is_a_test(pattern)]
+        tests = [test for pattern in safe_patterns for test in tests if pattern in test]
+    return sorted(tests)
+
+
+def execute_tests(tests_to_execute, **kwargs):
+    objects = globals()
+    tests = [objects[name] for name in tests_to_execute]
+    for test in tests:
+        task_id = kwargs['task_id'] + test.__name__
+        kw = {k: v for k, v in kwargs.items() if k != 'task_id'}
+        test(task_id=task_id, **kw)
+        print("-" * 80)
+
+
+def main():
+    global CONCENT_MESSAGING_TIME
```

Comment:
Isn't it enough to use constants defined at the beginning of the file, and just set their values here ?

Reply:
to set a value of a global variable in a local scope you need to declare it:
`global CONCENT_MESSAGING_TIME`

Reply:
to use global for writing in a local scope, you have to declare it first; otherwise you will create local variable of the same name as global

Reply:
But if we are changing its value, is it still a constant?

I agree with @rwrzesien here. There should be no reason to use `global` here. Either it's a constant and we don't need to change it or it's not a constant and we should be passing it everywhere as a parameter (possibly inside `ProtocolConstants`).

Third option, which I'm against because I don't see a good reason for using global variables (that are not constants) here, we rename it to `concent_messaging_time` so that it's obvious it's not a constant.